acra-censor/common/common.go

```diff
@@ -150,3 +176,142 @@ func ParsePatterns(rawPatterns []string) ([]sqlparser.Statement, error) {
 	}
 	return outputPatterns, nil
 }
+
+// TrimStringToN trims query to N chars.
+func TrimStringToN(query string, n int) string {
+	if len(query) <= n {
+		return query
+	}
+	return query[:n]
+}
+
+// HandleRawSQLQuery returns a normalized (lowercases SQL commands) SQL string,
+// and redacted SQL string with the params stripped out for display.
+// Taken from sqlparser package
+func HandleRawSQLQuery(sql string) (normalizedQuery, redactedQuery string, parsedQuery sqlparser.Statement, err error) {
+	bv := map[string]*querypb.BindVariable{}
+	sqlStripped, _ := sqlparser.SplitMarginComments(sql)
+
+	// sometimes queries might have ; at the end, that should be stripped
+	sqlStripped = strings.TrimSuffix(sqlStripped, ";")
+
+	stmt, err := sqlparser.Parse(sqlStripped)
+	outputStmt, _ := sqlparser.Parse(sqlStripped)
```

Comment:
for what second parsing the same query?

Reply:
Because I need initial query (without redacted values) in parsed form outside

Reply:
then better to parse second time after error check because in worst case we will parse two times with errors without need

Reply:
Agree